### PR TITLE
Feature/minor changes

### PR DESCRIPTION
**Pull request by vbkaisetsu**
This branch contains the following changes:

* Raises `TypeError` exception with a message when `__init__` is called twice in some classes.
* Adds `utils.set_encoding` function that can change encoding for passing string data to C++ functions.
* Adds `__repr__` special method to `Shape` class. It is useful in REPL interface.

```python
In [1]: from primitiv import Shape
In [2]: s = Shape([4,5,6,], 3)
In [3]: s
Out[3]: Shape([4,5,6]x3)
```

### CODE REVIEW

**Comment by odashi** [comment]
`Shape.__repr__` should return a correct Python expression that reproduces the same `Shape`. I.e.:
```python
>>> s = Shape([4,5,6], 3)
>>> s
Shape([4,5,6], 3)
>>> s = Shape()
>>> s
Shape([], 1)
```

**Comment by odashi** [comment]
`primitiv.utils` -> `primitiv.config` ?

**Comment by vbkaisetsu** [comment]
Thnaks. Fixed.

File: python-primitiv/primitiv/config.pyx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+cdef str py_primitiv_utils_encoding = None
+
+
+def set_encoding(str encoding = None):
+    global py_primitiv_utils_encoding
+    if encoding:
```
Comment (odashi) [inline python-primitiv/primitiv/config.pyx]: `is not None?`

File: python-primitiv/primitiv/config.pyx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        py_primitiv_utils_encoding = encoding
+    else:
+        import locale
+        lang, py_primitiv_utils_encoding = locale.getdefaultlocale()
```
Comment (odashi) [inline python-primitiv/primitiv/config.pyx]: `lang` is not used and could be replaced to `_`.

File: python-primitiv/primitiv/__init__.py [hunk 3]
```diff
@@ -12,6 +12,9 @@
 from primitiv._operator import _operators as operators
 from primitiv._operator import _tensor_operators as tensor_operators
 from primitiv import trainers
+from primitiv import config
+
+config.set_encoding()
```
Comment (odashi) [inline python-primitiv/primitiv/__init__.py]: Better to comment what this line do.

**Comment by vbkaisetsu** [comment]
Thanks. Fixed.

**Pull request merged**